### PR TITLE
[exporter/azuremonitor] Enhance AAD Documentation.

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
**Description:** <Describe what has changed.>

* Update AAD documentation to use connection string instead of instrumentation key. Follow up to #28854
* Modified the ingestion version from 2.0 to 2.1

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

Existing tests.

Output from manual run

``` json
--------- Transmitting 30 items ---------       {"kind": "exporter", "data_type": "logs", "name": "azuremonitor"}
2023-11-13T10:50:23.886-0800    debug   azuremonitorexporter@v0.88.0/factory.go:139     Telemetry transmitted in 378.439395ms   {"kind": "exporter", "data_type": "logs", "name": "azuremonitor"}
2023-11-13T10:50:23.886-0800    debug   azuremonitorexporter@v0.88.0/factory.go:139     Response: 200   {"kind": "exporter", "data_type": "logs", "name": "azuremonitor"}
2023-11-13T10:50:23.886-0800    debug   azuremonitorexporter@v0.88.0/factory.go:139     Items accepted/received: 30/30 {"kind": "exporter", "data_type": "logs", "name": "azuremonitor"}
```

**Documentation:** <Describe the documentation added.>
* Updated Authentication.md

### CODE REVIEW

File: exporter/azuremonitorexporter/AUTHENTICATION.md
Comment (crobert-1) [inline exporter/azuremonitorexporter/AUTHENTICATION.md:25]:
```suggestion
Local Authentication can be disabled in [Application Insights](https://learn.microsoft.com/en-us/azure/azure-monitor/app/azure-ad-authentication) and an AAD based identity can be used in conjunction with the instrumentation key. The instrumentation key, while still required by the Exporter in the connection string, is no longer solely sufficient to be able to import data.
```
The wording in the last sentence is a bit unclear. I'm happy to accept a different wording, but the way it's written now is unclear (to me).

File: exporter/azuremonitorexporter/connection_string_parser.go
Comment (crobert-1) [inline exporter/azuremonitorexporter/connection_string_parser.go:78]: What's the relevance of this change to the documentation update? If this is a functionality change it may be good to have a separate PR.

Comment (rajkumar-rangaraj) [reply exporter/azuremonitorexporter/connection_string_parser.go:78]: There is no separate document specifically addressing Application Insights versions. However, the Azure Monitor service schema, which contains the details, can be found here: [Service Schema](https://github.com/Azure/azure-rest-api-specs/blob/6013319f5a9f2fc613f762510e9dd46403b922bf/specification/applicationinsights/data-plane/Monitor.Exporters/preview/v2.1/swagger.json#L9)

This topic is also discussed in the troubleshooting document: [Troubleshooting Document](https://learn.microsoft.com/en-us/azure/azure-monitor/app/azure-ad-authentication?tabs=net#http11-400-authentication-not-supported)

This is a follow-up to work completed in PR #28854. The Instrumentation Key will soon be deprecated, along with specifying the endpoint. This PR includes a documentation update and transfers the responsibility of specifying the service version to the code itself. This approach has been adopted for Azure Monitor exporters in .NET, Java, Node.js, and Python.

I have included both the documentation and this change in the same PR, as they are closely related. For Azure Active Directory (AAD) authentication, it is mandatory to use v2.1, as v2.0 will not be supported. Previous documentation required customers to hard-code this section.

Reference: On March 31, 2025, support for instrumentation key ingestion will end. Instrumentation key ingestion will continue to work, but we'll no longer provide updates or support for the feature. [Transition to connection strings](https://learn.microsoft.com/en-in/azure/azure-monitor/app/migrate-from-instrumentation-keys-to-connection-strings) to take advantage of [new capabilities](https://learn.microsoft.com/en-in/azure/azure-monitor/app/migrate-from-instrumentation-keys-to-connection-strings#new-capabilities).

File: exporter/azuremonitorexporter/AUTHENTICATION.md
Comment (rajkumar-rangaraj) [reply exporter/azuremonitorexporter/AUTHENTICATION.md]: I agree, this sentence itself is not needed. It is reiterating the previous statement in a different way.

**Review by pcwiese — APPROVED** [review]

**Review by crobert-1 — APPROVED** [review]

**Review by atoulme — APPROVED** [review]

**Comment by rajkumar-rangaraj** [comment]
@open-telemetry/collector-contrib-approvers Could you please help merge this PR, got an approval from the code owner.

**Review by evan-bradley — APPROVED** [review]

**Comment by evan-bradley** [comment]
@rajkumar-rangaraj We're currently performing a release. Once the release is over, I'll merge this.

File: exporter/azuremonitorexporter/connection_string_parser.go
Comment (codeboten) [reply exporter/azuremonitorexporter/connection_string_parser.go:78]: this is changing the behaviour though, should it not be documented in a changelog entry for end users?

Comment (rajkumar-rangaraj) [reply exporter/azuremonitorexporter/connection_string_parser.go:78]: Added changelog entry.

**Pull request merged**
